src/AnalyzerPowerPack/Core/Usage/SerializationRulesDiagnosticAnalyzer.cs

```diff
@@ -186,7 +234,34 @@ public void AnalyzeSymbol(SymbolAnalysisContext context)
 
             private bool IsSerializable(ITypeSymbol namedTypeSymbol)
             {
-                return namedTypeSymbol.GetAttributes().Any(a => a.AttributeClass == _serializableAttributeTypeSymbol);
+                return IsPrimitiveType(namedTypeSymbol) ||
+                       namedTypeSymbol.SpecialType == SpecialType.System_String ||
+                       namedTypeSymbol.GetAttributes()
+                           .Any(a => a.AttributeClass.Equals(_serializableAttributeTypeSymbol));
+            }
+
+            private static bool IsPrimitiveType(ITypeSymbol type)
+            {
+                switch (type.SpecialType)
+                {
+                    case SpecialType.System_Boolean:
+                    case SpecialType.System_Byte:
```

Comment:
In that case it has to be fixed here as well:
http://source.roslyn.io/#System.Runtime.Analyzers/Reliability/DoNotLockOnObjectsWithWeakIdentity.cs,92432dae4f408c01